lib/util.js

```diff
@@ -59,32 +63,44 @@ exports.getLabels = function (labelNames, args) {
 	return acc;
 };
 
-function hashObject(labels) {
-	// We don't actually need a hash here. We just need a string that
-	// is unique for each possible labels object and consistent across
-	// calls with equivalent labels objects.
-	let keys = Object.keys(labels);
+function fastHashObject(keys, labels) {
 	if (keys.length === 0) {
 		return '';
 	}
-	// else
-	if (keys.length > 1) {
-		keys = keys.sort(); // need consistency across calls
-	}
 
 	let hash = '';
-	let i = 0;
-	const size = keys.length;
-	for (; i < size - 1; i++) {
-		hash += `${keys[i]}:${labels[keys[i]]},`;
+
+	for (let i = 0; i < keys.length; i++) {
+		const key = keys[i];
+		const value = labels[key];
+		if (typeof value === 'undefined') continue;
```

Comment:
is `typeof value === 'undefined'` faster than just `value === undefined`? One cannot redefine `undefined` in a strict context afaik, so it should be needed to workaround

Reply:
I think it's hard in benchmarks to see the difference between the two options here, I think it falls within a margin of error.

I ran the benchmarks with `typeof` and `=== undefined` and I am getting sometimes that `=== undefined` is faster and sometimes it's slower. It's probably because I am running the benchmarks on my laptop and there is lots of variance.

I looked at difference of v8 bytecode and looks like `=== undefined` have optimized bytecode:
```
typeof x === 'undefined'
         0x222300199ff0 @   40 : 20 05             TestTypeOf #5
         0x222300199ff2 @   42 : 9a 04             JumpIfFalse [4] (0x222300199ff6 @ 46)
         0x222300199ff4 @   44 : 8b 1f             Jump [31] (0x22230019a013 @ 75)

x === undefined
         0x2c0a00199ff0 @   40 : 9e 04             JumpIfNotUndefined [4] (0x2c0a00199ff4 @ 44)
```

And I see that lodash is doing `=== undefined`.

So I am changing to `=== undefined` check.